# IMSC 2024 Short Course: Computational and Statistical Proteomic Analysis
The growth of biological MS deepens the need to understand computation and statistics; these skills are essential for the interpretation of proteomic studies. Designed for mass spectrometrists at an intermediate level, this course incorporates real-world use cases for the computational tools used to identify and quantify proteins/peptides. This tutorial will emphasize the design of well-powered studies, the conceptional frameworks associated with the identification of peptides and PTMs, and the detection and quantitation of biological differences. Participants will be introduced to methods for visualising and interpreting datasets, including statistical techniques using common bioinformatics tools such as the R statistical environment. At the conclusion of this course participants should have a working understanding of the key considerations associated with computational analysis of proteomic datasets.

Proteomics plays a crucial role in understanding cellular processes and disease mechanisms. Analyzing proteomic data involves several key steps, including data preprocessing, quality assessment, normalization, statistical testing, and visualization. This GitHub Repository provides a comprehensive guide to these steps, demonstrating how to handle and interpret proteomic datasets using R.

## Session 0: Installing R and R Studio
RStudio is an integrated development environment (IDE) for R and Python. It includes a console, syntax-highlighting editor that supports direct code execution, and tools for plotting, history, debugging, and workspace management. RStudio is available in open source and commercial editions and runs on the desktop (Windows, Mac, and Linux).

### Download and Install R
[Download R for Linux](https://cran.rstudio.com/bin/linux/)<br>
[Download R for macOS](https://cran.rstudio.com/bin/macosx/)<br>
[Download R for Windows](https://cran.rstudio.com/bin/windows/)<br>

### Install RStudio
[Download RStudio Desktop for Linux](https://posit.co/download/rstudio-desktop/)<br>
[Download RStudio Desktop for macOS](https://download1.rstudio.org/electron/macos/RStudio-2024.04.2-764.dmg)<br>
[Download RStudio Desktop for Windows](https://download1.rstudio.org/electron/windows/RStudio-2024.04.2-764.exe)<br>

[RStudio IDE User Guide](https://docs.posit.co/ide/user/)<br>

### Install Packages Required for this Course
Download and run the [Install Packages](https://github.com/JoBBurt/IMSC-2024-Computational-and-Statistical-Proteomic-Analysis/blob/main/Session_0/Install%20Packages.Rmd) markdown file. Please note that this takes about 45 minutes to run on an M1 MacBook Air and requires active user input to complete.

## Session 1: Quality Control Assessment
In this session, we will explore quality control (QC) assessment for proteomic data. We will be using data from the paper ["Substantial downregulation of mitochondrial and peroxisomal proteins during acute kidney injury revealed by data-independent acquisition proteomics"](https://doi.org/10.1002/pmic.202300162).

![image](https://github.com/user-attachments/assets/830b6e0b-5ccd-44a7-83df-2930dec926d4)


## Session 2: Comparative Analysis of Proteomic Data


## Session 3: Gene Ontology and Network Analysis


## Useful Resources
[Building a Bioinformaticist by David Tabb, PhD](https://pickingupthetabb.wordpress.com/building-a-bioinformaticist/free-online-training-in-bioinformatics-and-biostatistics/)<br>
[May Institute: Computation and Statistics for Mass Spectrometry and Proteomics by Olga Vitek, PhD](https://www.youtube.com/@MayInstituteNEU)<br>
[Analysis of Proteomics Data by Brett Phinney, PhD](https://www.youtube.com/@UCDavisProteomics)<br>

## Licensing

[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)<br>
All source code is licensed under the Apache License 2.0. This includes the notebooks deployed via Google Colaboratory.

<a rel="license" href="http://creativecommons.org/licenses/by-sa/4.0/"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-sa/4.0/80x15.png" /></a><br />Content and artwork is licensed under a Creative Commons Attribution-ShareAlike 4.0 International License.
