---
title: "Comparative Analysis and Gene Ontology of Proteomic Data"
author: "Jordan Burton"
date: "2024-08-08"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
# Conmpariative Analysis of Proteomic Data

## Introduction
In this section, we will explore the use of comparisons generated from Spectronaut or the Limma R package for statistical testing.

## Setup
Before diving into the analysis, we need to ensure that all required packages are installed and loaded. The following code chunk installs and loads the necessary libraries for our analysis.

```{r Install Packages, include=FALSE, eval = F}
# Install packages required for analysis
install.packages("tidyverse")
install.packages("scales")
install.packages("stringr")
install.packages("ggplot2")
install.packages("gplots")
install.packages("ggrepel")
install.packages("corrplot")
install.packages("VennDiagram")
install.packages('hrbrthemes')
# Install BiocManager and Related Packages
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
```

```{r Load Packages}
# Load libraries required for analysis
library(tidyverse)
library(scales)
library(stringr)
library(ggplot2)
library(gplots)
library(ggrepel)
library(corrplot)
library(VennDiagram)
library(hrbrthemes)

theme_set(theme_bw(base_size = 18))
```

```{r Set Theme}
# Set the Theme
theme_science <- function(base_size = 12) {
  theme_bw(base_size = base_size) +
    theme(
      text = element_text(family = "Arial"),  # Set font family to serif
      plot.title = element_text(size = 14, face = "bold", hjust = 0.5),
      axis.title = element_text(size = 12, color = "black"),  # Set axis title color to black
      axis.text = element_text(size = 10, color = "black"),  # Set axis text color to black
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      panel.border = element_blank(),
      axis.line = element_line(color = "black")
    ) 
}
```

## Study Overview

In this study, we investigate the APOE3 allele, which is a neutral allele that does not increase the risk for AD, in neurons and microglia to understand the baseline functions of APOE in the brain. To characterize the cells' proteomes, we developed a sensitive data-independent acquisition (DIA) mass spectrometry workflow using a SCIEX ZenoTOF 7600 system coupled to a Waters M-Class UPLC. There are 6 biological replicates for each cell type and three technical replicates were acquired per biological replicate.

```{r Load protein data from spectronaut}
protein <- read.csv("data/23_0802_LMF1_MSan2_3rep_combined_panHuman_Report_Birgit_Protein Quant_Pivot (Pivot).csv",
                    stringsAsFactors = FALSE)

# Rearrange Protein Column Order
pro_colorder <- c("PG.Genes", "PG.ProteinDescriptions","PG.Qvalue",
              "PG.ProteinNames",
              "PG.UniProtIds", "PG.BiologicalProcess",
              "PG.CellularComponent", "PG.MolecularFunction", names(protein[,9:ncol(protein)]))
protein <- protein[,pro_colorder]
print(head(protein))
nrow(protein)
```

Next, we prepare a matrix of protein quantities for analysis. This involves selecting relevant columns, renaming them for simplicity, and setting the row names to gene names. LMF = Neurons, MSan = Microglia.

```{r Prepare a matrix of the protein data for analysis}
protein_mat <- data.matrix(dplyr::select(protein, contains(c("PG.Quantity"))))
conditions <- read.csv("data/23_0802_LMF1_MSan2_3rep_combined_panHuman_ConditionSetup.csv",
                       stringsAsFactors = F)
# Rename columns in protein data matrix to simplify names for plotting
colnames(protein_mat)
names <- conditions$Names
colnames(protein_mat) <- names

# Rename rows in protein data matrix to gene names
rownames(protein_mat) <- protein$PG.Genes

# Inspect the matrix
print(head(protein_mat))
```

## Pairwise Comparisons
To understand the differences between conditions, we perform pairwise comparisons of protein abundances. Here, we load the comparison data from Spectronaut and rearrange the columns for better readability. This step helps in identifying differentially expressed proteins between the groups.

## Comparison data from Spectronaut

```{r Load Pairwise Comparison Data from Spectronaut}
comparisons <- read.csv("data/23_0802_LMF1_MSan2_3rep_combined_panHuman_candidates.csv",
                        stringsAsFactors = FALSE)

# Rearrange Comparison Column Order
comp_colorder <- c("Comparison..group1.group2.", "Genes", "ProteinDescriptions",
              "ProteinNames", "UniProtIds", "ProteinGroups", "Group",
              "AVG.Log2.Ratio", "Qvalue", "Absolute.AVG.Log2.Ratio",
              "Pvalue", "X..of.Ratios",
              names(comparisons[,13:ncol(comparisons)]))
comparisons <- comparisons[,comp_colorder]
print(head(comparisons))
nrow(comparisons)
```

Histograms of p-values and fold changes provide insights into the distribution of these statistical measures. P-values indicate the significance of the differences observed, while fold changes represent the magnitude of these differences.

```{r Histogram of p-Values and Fold Change}
par(mfrow=c(1,2))
hist(comparisons$Qvalue, main = "q-Value Histogram", xlab = 'q-Value')
hist(comparisons$AVG.Log2.Ratio, main = "Log2(Fold Change) Histogram", xlab = 'Log2(FC)')
```

A volcano plot combines p-values and fold changes to highlight significantly different proteins. Proteins with large fold changes and small p-values are considered significant and are color-coded for easy identification.

```{r Volcano Plot}
# Add additional column in comparisons for plotting
comparisons$minuslogqval <- -1*log10(comparisons$Qvalue)
# Assign colors for significantly up and down regulated proteins, nonregulated proteins
mycolors <- c("Blue", "Red", "Gray")
names(mycolors) <- c("Blue", "Red", "Gray")

# Assign colors to significantly altered proteins
df <- comparisons
df$Color <- ifelse(df$AVG.Log2.Ratio >= 0.58 & df$Qvalue < 0.001, "Red", "Grey")
df$Color <- ifelse(df$AVG.Log2.Ratio <= -0.58 & df$Qvalue < 0.001, "Blue", df$Color)
down <- paste(count(df[df$Color == "Blue",]), "Down", sep = " ")
up <- paste(count(df[df$Color == "Red",]), "Up", sep = " ")

# Set the y-axis
max_minuslogqval <- ceiling(max(-log10(df$Qvalue), na.rm = TRUE))
# Q-value = 0 so max_minuslogqval is Infinite
max_minuslogqval <- -log10(8.3E-321)

ggplot(data = df, aes(x = AVG.Log2.Ratio, y = minuslogqval, col = Color, text = Genes)) +
    geom_point() +
    geom_vline(xintercept = c(-0.58, 0.58), col = "black", linetype = "dashed") +
    geom_hline(yintercept = -log10(0.001), col = "black", linetype = "dashed") +
    scale_color_manual(values = mycolors) +
    ylab("-Log10(q-value)") +
    scale_x_continuous(name = "Log2(fold change)", limits = c(-4.5,8),
                      labels = c(-4,-2,0,2,4,6), breaks = c(-4,-2,0,2,4,6)) +
    ylim(c(0, max_minuslogqval))+
    theme_classic() +
    theme(axis.title =element_text(size = 20, color = "black"),
          axis.text = element_text(size = 18, color = "black"),
          legend.position = "none") +
    ggtitle("Microglia vs. Neurons",
      paste(nrow(df), "Total Proteins: ",down,"&", up, "Rgeulated Proteins"))

```

## Using the Limma package for pairwise comparisons

The Bioconductor R package limma (Linear Models for Microarray Data) is a powerful and widely used tool for analyzing gene expression data from microarray and RNA-seq experiments. It provides methods for fitting linear models to microarray data, making it possible to identify differentially expressed genes, perform statistical tests, and correct for multiple testing.

### Key Features of limma

	1.	Linear Models:
limma uses linear models to assess differential expression. This approach allows for flexible experimental designs and the inclusion of multiple factors and covariates.
	2.	Empirical Bayes Moderation:
limma applies an empirical Bayes approach to shrink the estimated variances of the coefficients, which improves the reliability of the statistical tests, especially when sample sizes are small.
	3.	Differential Expression Analysis:
The package provides functions to identify differentially expressed genes between conditions or groups. It supports complex experimental designs, including paired samples and time-course experiments.
	4.	Normalization:
limma includes methods for normalizing microarray data to correct for technical variations and ensure that the observed differences are due to biological variation.
	5.	Annotation:
limma can be used in conjunction with annotation packages to add biological context to the results, such as gene ontology terms and pathway information.
	6.	Proteomics Analysis:
While initially designed for microarray data, limma has been extended to handle proteomics data, offering tools for differential expression analysis of abundance or count data using the voom transformation.

### Practical Applications

limma is particularly useful in the following scenarios:

	•	Identifying Differentially Expressed Genes: Discovering genes that show significant changes in expression between experimental conditions or groups.
	•	Complex Experimental Designs: Analyzing data from experiments with multiple factors, covariates, or paired samples.
	•	High-Throughput Data: Handling large-scale gene expression data from microarrays and RNA-seq experiments.
	•	Statistical Rigor: Using empirical Bayes moderation to improve the reliability of results, especially when dealing with small sample sizes.

By providing a robust framework for differential expression analysis, limma enables researchers to uncover meaningful biological insights from high-throughput gene expression data and to interpret these findings in the context of their experimental designs.

```{r install limma, eval = F}
BiocManager::install("limma")
```

```{r load limma}
library(limma)
```


A design matrix is created to specify the experimental design and the factors to be compared. In this case, the matrix indicates the different treatments applied to the samples.

```{r Create a design matrix}
treatment <- as.factor(conditions$Condition)
design <- model.matrix(~0 + treatment)
colnames(design) <- levels(treatment)

# Print the design matrix to verify
print(design)
```

Using the design matrix, we fit a linear model to the protein data and perform pairwise comparisons. This step involves calculating contrasts and applying empirical Bayes moderation to obtain more stable estimates of the fold changes and p-values.

```{r Limma linear modeling and pairwise comparisons}
# Fit the linear model
fit <- lmFit(protein_mat, design)

# Create a contrast matrix for the comparison of interest
contrast_matrix <- makeContrasts(Microglia_vs_Neurons = MSan2 - LMF1, levels = design)

# Fit the contrasts
fit2 <- contrasts.fit(fit, contrast_matrix)
fit2 <- eBayes(fit2)

# Get the results
results <- topTable(fit2, adjust = "fdr", number = Inf)

# Print the results
print(head(results))
```

Similar to the previous analysis, we create histograms to visualize the distribution of p-values and fold changes in the Limma results.

```{r Limma: Histogram of p-Values and Fold Change}
par(mfrow=c(1,2))
hist(results$adj.P.Val, main = "q-Value Histogram", xlab = 'q-Value')
hist(results$logFC, main = "Log2(Fold Change) Histogram", xlab = 'Log2(FC)')
```

The volcano plot for Limma results shows significantly different proteins based on their fold changes and adjusted p-values. This visualization helps in identifying the most relevant proteins for further analysis.

```{r Limma: Volcano Plot}

# Add additional column in comparisons for plotting
results$minuslogqval <- -1*log10(results$adj.P.Val)
# Assign colors for significantly up and down regulated proteins, nonregulated proteins
mycolors <- c("Blue", "Red", "Gray")
names(mycolors) <- c("Blue", "Red", "Gray")

# Assign colors to significantly altered proteins
df <- results
df$Color <- ifelse(df$logFC >= 0.58 & df$adj.P.Val < 0.001, "Red", "Grey")
df$Color <- ifelse(df$logFC <= -0.58 & df$adj.P.Val < 0.001, "Blue", df$Color)
down <- paste(count(df[df$Color == "Blue",]), "Down", sep = " ")
up <- paste(count(df[df$Color == "Red",]), "Up", sep = " ")

# Set the y-axis
max_minuslogqval <- ceiling(max(-log10(df$adj.P.Val), na.rm = TRUE))

ggplot(data = df, aes(x = logFC, y = minuslogqval, col = Color, text = ID)) +
    geom_point() +
    geom_vline(xintercept = c(-0.58, 0.58), col = "black", linetype = "dashed") +
    geom_hline(yintercept = -log10(0.001), col = "black", linetype = "dashed") +
    scale_color_manual(values = mycolors) +
    ylab("-Log10(q-value)") +
    scale_x_continuous(name = "Log2(fold change)", limits = c(-100,100),
                     labels = c(-100,-75,-50,-25,0,25,50,75,100), 
                     breaks = c(-100,-75,-50,-25,0,25,50,75,100)) +
    ylim(c(0, max_minuslogqval))+
    theme_classic() +
    theme(axis.title =element_text(size = 20, color = "black"),
          axis.text = element_text(size = 18, color = "black"),
          legend.position = "none") +
    ggtitle("Microglia vs. Neurons",
      paste(nrow(df), "Total Proteins: ",
            down,"&", up, "Rgeulated Proteins"))

```

## Overlap between Spectronaut and Limma comparisons
To identify common proteins between the Spectronaut and Limma analyses, we create a Venn diagram. This diagram shows the overlap of significantly altered proteins, highlighting those consistently identified across both methods.

```{r Pull out Singificantly Altered Proteins from Spectronaut and Limma}
# Subset out Significantly Altered Proteins
SN_sign_results <- subset(comparisons,
      Absolute.AVG.Log2.Ratio > 0.58 & Qvalue < 0.001)
SN_up <- subset(SN_sign_results, AVG.Log2.Ratio > 0)
SN_down <- subset(SN_sign_results, AVG.Log2.Ratio < 0)

Lim_sign_results <- subset(results, abs(logFC) > 0.58 & adj.P.Val < 0.001)
Lim_up <- subset(Lim_sign_results, logFC > 0)
Lim_down <- subset(Lim_sign_results, logFC < 0)

# Get the protein names
SN_up_proteins <- SN_up$Genes
SN_down_proteins <- SN_down$Genes
Lim_up_proteins <- Lim_up$ID
Lim_down_proteins <- Lim_down$ID
```

# Pathway Analysis and Protein Localization
## Gene Ontology Analysis
Gene Ontology (GO) analysis is a powerful tool to interpret the biological significance of protein sets by categorizing them into predefined biological processes, molecular functions, and cellular components. 

### clusterProfiler

clusterProfiler is a versatile package designed for statistical analysis and visualization of functional profiles for genes and gene clusters. It supports a wide range of analyses, including gene ontology (GO) enrichment analysis, Kyoto Encyclopedia of Genes and Genomes (KEGG) pathway enrichment analysis, and more.

#### Key Features

	1.	Gene Ontology (GO) Enrichment Analysis:
clusterProfiler provides tools for GO enrichment analysis to identify over-represented GO terms in a set of genes.
	2.	KEGG Pathway Enrichment Analysis:
It supports KEGG pathway enrichment analysis, allowing users to identify pathways significantly associated with their gene sets.
	3.	Visualization:
The package includes various visualization methods, such as bar plots, dot plots, cnet plots (category network plots), and enrich maps, to help interpret enrichment results.
	4.	Support for Multiple Organisms:
clusterProfiler can be used with various organisms, and users can specify different databases for annotation.
	5.	GSEA (Gene Set Enrichment Analysis):
clusterProfiler also supports gene set enrichment analysis, which ranks genes and tests whether predefined gene sets are significantly enriched at the top or bottom of the ranked list.

### org.Hs.eg.db

org.Hs.eg.db is an annotation package that provides comprehensive gene-centric annotations for human ontologies. This package is part of the Bioconductor annotation packages and contains mappings between various identifiers, such as Entrez Gene IDs, Ensembl IDs, gene symbols, and more.

#### Key Features

	1.	Comprehensive Annotation:
org.Hs.eg.db includes extensive annotations, covering gene ontology terms, KEGG pathways, chromosome locations, and other relevant biological data.
	2.	Multiple Identifier Support:
The package supports a variety of gene identifiers, making it versatile for different types of analyses.
	3.	Integration with Other Packages:
org.Hs.eg.db can be seamlessly integrated with other Bioconductor packages like clusterProfiler, making it easy to perform functional analyses and annotations.

### Practical Applications

The combination of clusterProfiler and org.HS.eg.db is particularly useful in the following scenarios:

	•	Functional Enrichment Analysis: Identifying enriched GO terms, KEGG pathways, and other functional categories in a set of differentially expressed genes.
	•	Gene Set Analysis: Conducting gene set enrichment analysis to determine whether predefined sets of genes show statistically significant differences in expression.
	•	Gene Annotation: Annotating gene lists with relevant biological information, including gene symbols, GO terms, and pathway associations.

These tools enable researchers to gain deeper insights into the biological significance of their high-throughput genomic data and to uncover the molecular mechanisms underlying their experimental observations.

```{r Install Packages for Gene Ontology Analysis, eval=F}
BiocManager::install("clusterProfiler")
BiocManager::install("org.Hs.eg.db") # human database
#BiocManager::install("org.Mm.eg.db") # mouse database
```

```{r Load Libraries for Gene Ontology Analysis}
library(clusterProfiler)
library(org.Hs.eg.db)
#library(org.Mm.eg.db)
```

Next, we convert protein names to Entrez IDs, which are required for GO analysis. This step involves mapping protein symbols to their corresponding Entrez Gene IDs using the bitr function.

```{r Convert protein names to Entrez IDs for GO analysis}
# Here assuming the protein names are in a format directly convertible. You might need a mapping step in real cases.
SN_up_entrez <- bitr(SN_up_proteins, fromType="SYMBOL",
                            toType="ENTREZID", OrgDb="org.Hs.eg.db")$ENTREZID
SN_down_entrez <- bitr(SN_down_proteins, fromType="SYMBOL",
                            toType="ENTREZID", OrgDb="org.Hs.eg.db")$ENTREZID
Limma_up_entrez <- bitr(Lim_up_proteins, fromType="SYMBOL",
                            toType="ENTREZID", OrgDb="org.Hs.eg.db")$ENTREZID
Limma_down_entrez <- bitr(Lim_down_proteins, fromType="SYMBOL",
                            toType="ENTREZID", OrgDb="org.Hs.eg.db")$ENTREZID
```

We then perform GO enrichment analysis for each set of proteins to identify overrepresented biological processes. The enrichGO function helps in this analysis.

```{r Perform GO enrichment analysis for each set of proteins}
go_SN_up <- enrichGO(gene = SN_up_entrez, OrgDb = org.Hs.eg.db,
          ont = "BP", pAdjustMethod = "BH", qvalueCutoff = 0.05, readable = TRUE)

go_SN_down <- enrichGO(gene = SN_down_entrez, OrgDb = org.Hs.eg.db,
          ont = "BP", pAdjustMethod = "BH", qvalueCutoff = 0.05, readable = TRUE)

go_Limma_up <- enrichGO(gene = Limma_up_entrez, OrgDb = org.Hs.eg.db,
          ont = "BP", pAdjustMethod = "BH", qvalueCutoff = 0.05, readable = TRUE)

go_Limma_down <- enrichGO(gene = Limma_down_entrez, OrgDb = org.Hs.eg.db,
          ont = "BP", pAdjustMethod = "BH", qvalueCutoff = 0.05, readable = TRUE)
```

```{r Dotplots of Biological Processes for Spectronaut}
par(mfrow=c(1,2))
dotplot(go_SN_up, showCategory=10) + 
  ggtitle("GO Enrichment for Upregulated Proteins\nin the Microglia vs Neurons Comparison") +
  theme_science()
dotplot(go_SN_down, showCategory=10) + 
  ggtitle("GO Enrichment for Downregulated Proteins\nin the Microglia vs Neurons Comparison") +
  theme_science()
```

```{r Dotplots of Biological Processes for Limma}
par(mfrow=c(1,2))
dotplot(go_Limma_up, showCategory=10) + 
  ggtitle("GO Enrichment for Upregulated Proteins\nin the Microglia vs Neurons Comparison") + 
  theme_science()
dotplot(go_Limma_down, showCategory=10) + 
  ggtitle("GO Enrichment for Downregulated Proteins\nin the Microglia vs Neurons Comparison") +
  theme_science()
```

## Localization of Organellar Proteins

Organellar protein localization helps in understanding the subcellular distribution of proteins. The Localization of Organelle Proteins by Isotope Tagging (LOPIT) technique is used to map proteins to specific cellular compartments. We start by installing and loading the necessary packages.

```{r Install Packages for LOPIT, eval=F}
install.packages("reshape2")
install.packages("Rtsne")
install.packages("RColorBrewer")
install.packages("ggpubr")
BiocManager::install("pRoloc")
BiocManager::install("pRolocdata")
```

```{r Load Packages for LOPIT}
library("reshape2")
library("Rtsne")
library("RColorBrewer")
library("ggpubr")
library("pRoloc")
library("pRolocdata")
```

We set default themes for LOPIT plots to ensure consistency in the visual presentation.

```{r LOPIT plot defaults}
theme_set(theme_bw(base_size = 12))
update_geom_defaults("point", list(size = 0.7))
```

Using data from Thul et al. 2017, we generate a t-SNE plot to visualize the subcellular localization of proteins. This plot helps in identifying the distribution patterns of proteins across different organelles.

```{r Mouse Reference LOPIT Plot}
# Load data from Thul et. al. 2017
data("hyperLOPITU2OS2017")
  
# use T-SNE algorithm to generate plot dimensions and assign points to dimensions
set.seed(11)
p <- plot2D(hyperLOPITU2OS2017, method = "t-SNE")
  
# Formatting the data 
u2os <- data.frame(Accession = row.names(p), p, assignment = as.character(fData(hyperLOPITU2OS2017)$assignment), stringsAsFactors = FALSE)
u2os$Accession <- sapply(strsplit(u2os$Accession, "-"),'[', 1)
  
## Add assignment name to plot
textFrame <- data.frame(x = c(30, 8, -25, -40, 38, -27, -31, 52, -25, 28, 14, 25), 
                        y = c(22, 30, 35, 14, 3, -30, -12, 0, -2, 8, -28, -23), 
                        text = c("Cytosol", "ER", "Golgi", "Lysosome", "Mitochondria", 
                                 "Nucleus", "Nucleus-Chromatin", "Peroxisome", "Plasma Membrane", 
                                 "Proteasome", "40S Ribosome", "60S Ribosome"))
  
# plotting the hyperlopit data
hyperLOPIT <- u2os %>%
   mutate(annotated = assignment != "unknown") %>%
   ggplot(aes(x = Dimension.1, y = Dimension.2)) +
   geom_point(data = function(x){x[!(x$annotated), ]}, color = grey(0.9)) +
   geom_point(data = function(x){x[(x$annotated), ]}, aes(color = assignment)) +
   geom_text(data = textFrame, aes(x = x, y = y, label = text), size = 3.5) +
   scale_color_manual(values = c(brewer.pal(12, "Paired"))) +
   labs(color = "Localization", x = "t-SNE Dim. 1", y = "t-SNE Dim. 2") +
   theme(axis.text.x = element_blank(),
         axis.text.y = element_blank(),
         axis.ticks = element_blank(), 
         legend.position = 'none')

# Plot the Lopit Reference Plot
hyperLOPIT
```

Next, we overlay the reference plot with fold change data from pairwise comparisons. This helps in visualizing the differences in protein abundance across conditions within the context of subcellular localization.

```{r Overlay reference plot with fold change data from pairwise comparisons}
## Fold Change Plots
df <- comparisons
df$minuslogqval <- -1*log10(df$Qvalue)
df$Color <- ifelse(df$AVG.Log2.Ratio >= 0.58 & df$Qvalue < 0.001, "Red", "Grey")
df$Color <- ifelse(df$AVG.Log2.Ratio <= -0.58 & df$Qvalue < 0.001, "Blue", df$Color)
df$Label <- NA
u2os$inCan <- u2os$Accession %in% df$UniProtIds #1721 proteins
u2osCan <- df %>%
  full_join(u2os, by = c("UniProtIds" = "Accession"))
    
textFrame <- data.frame(x = c(-5, 5, -15, -33, 28, 7, 7, -5, 40, 25, 0, 17, -20, -28), 
                       y = c(18, 5, -6, -2, -20, -39.5, -26, -41, 4, 25, 45, -3, -31, 23), 
                       text = c("40S R", "60S R", "AC", "Cyt", "ER/GA", "End", "EM", "Lys", "Mito", "Nuc-Chr", "Nuc Non-Chr", "Per", "PM", "Pro"))
# Filter out non-statistically significant data points
myCan <- u2osCan[!is.na(u2osCan$Absolute.AVG.Log2.Ratio),]
myCan <- myCan[myCan$Color != "Grey",]
    
canLOPIT <- u2osCan %>%
   mutate(identified = !is.na(u2osCan$inCan)) %>%
   ggplot(aes(x = Dimension.1, y = Dimension.2)) +
   geom_point(alpha = 0.1) +
   geom_point(data = myCan, aes(x = Dimension.1, y = Dimension.2, size = Absolute.AVG.Log2.Ratio),
              color = myCan$Color, alpha = 0.3) +
   #scale_size_manual(values = c(0.18, 0.375, .75, 1.5)) +
   labs(color = "Localization", title = x, x = "t-SNE Dim. 1", y = "t-SNE Dim. 2") +
   theme(#axis.text.x = element_blank(),
   #axis.text.y = element_blank(),
   #axis.ticks = element_blank(),
   legend.position.inside = c(0.88, 0.83)) +
   guides(size=guide_legend(title="|log2(FC)| Ratio")) +
   geom_text(data = textFrame, aes(x = x, y = y, label = text), size = 4.5)
    
# Plot the fold change LOPIT Plots    
canLOPIT
```

